js-envs/src/main/scala/org/scalajs/jsenv/rhino/LazyScalaJSScope.scala

```diff
@@ -56,18 +62,23 @@ class LazyScalaJSScope(
   override def getClassName() = "LazyScalaJSScope"
 
   override def get(name: String, start: Scriptable) = {
-    fields.getOrElse(name, {
-      try {
-        load(name)
-        fields.getOrElse(name, Scriptable.NOT_FOUND)
-      } catch {
-        // We need to re-throw the exception if `load` fails, otherwise the
-        // JavaScript runtime will not catch it.
-        case t: ScalaJSCoreLib.ClassNotFoundException =>
-          throw Context.throwAsScriptRuntimeEx(t)
-      }
-    }).asInstanceOf[AnyRef]
+    if (name == "__noSuchMethod__") {
```

Comment:
It seems Rhino likes to pretend it's cool because it has Ruby features ^^ Apparently, when calling a method that does not exist, instead of immediately throwing a TypeError, Rhino attempts to call `__noSuchMethod__`. Only after that does it fall back to the TypeError.

The problem is that our code throws a `ClassNotFoundException` early. Then apparently this also causes Rhino to call `__noSuchMethod__`. And *that* again throws a `ClassNotFoundException`. At this point, Rhino propagates the latter instead of the former, which means the exception that is actually thrown does not refer to the correct class name.

It seems this was not a problem with the earlier version of `load`/`loadAncestors`. But now it is. Explicitly returning `undefined` for `__noSuchMethod__` fixes the problem.